akka-docs/src/test/scala/docs/stream/operators/sink/HeadOption.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.stream.operators.sink
+
+import akka.actor.ActorSystem
+import akka.stream.scaladsl.{ Sink, Source }
+
+import scala.concurrent.{ ExecutionContextExecutor, Future }
+
+object HeadOption {
+  implicit val system: ActorSystem = ???
+  implicit val ec: ExecutionContextExecutor = system.dispatcher
+  def headOptionExample: Future[Unit] = {
+    //#headoption
+    val source = Source.empty
+    val result: Future[Option[Int]] = source.runWith(Sink.headOption)
+    result.map(println)
```

Comment:
Use `foreach` for sideffect only on Future

Reply:
You are suggesting to use foreach in place of map?

Reply:
Yes. `map` transforms a `Future[T]` to a `Future[U]` given a `T => U` function, but in this case we just want to react/side effect on future completion so it is somewhat non-idiomatic to use `map`, that's what `Future.foreach` is meant for.